ci.sh

```diff
@@ -37,11 +37,11 @@ python -c "import sys, struct, ssl; print('python:', sys.version); print('versio
 echo "::endgroup::"
 
 echo "::group::Install dependencies"
-python -m pip install -U pip setuptools wheel
+python -m pip install -U pip build
 python -m pip --version
 
-python setup.py sdist --formats=zip
-python -m pip install dist/*.zip
+python -m build --sdist
+python -m pip install dist/*.tar.gz
```

Comment:
I'd be tempted to do this:

```suggestion
python -m build
python -m pip install dist/*.whl
```

Reply:
Could even do `pip install .`, though it might be better to keep it separate to get better logs.

Reply:
`pip install .` does not detect problems with your MANIFEST.in

Reply:
What problems does `python -m build` detect that `pip install .` does not?

Reply:
Hm, I think this relates to building the sdist which IIRC pip install . does not do, yes?

Reply:
Building an sdist can omit files in the source tree missing from MANIFEST.in `pip install .` doesn't omit those files

Reply:
I was thoroughly confused by this response. Are you saying that `pip install .` skips parts of the build process somehow, or not? As you said "pip install . _doesn't_ omit those files".

Reply:
Yeah pip install . skips the sdist phase which is where MANIFEST.in is used

Reply:
I don't think this argument holds water since `python -m build` happily builds the sdist even when files included in `MANIFEST.in` are missing.

Reply:
Right but then those files will be missing from the install

Reply:
They'll be missing (without warnings) no matter how you do the build.

Reply:
If you do `pip install .` they won't be missing

Reply:
How can nonexistent files not be missing from the install?

Reply:
the point is that `pip install .` skips the sdist stage and we want to test the sdist stage

Reply:
Ok, so: `pip install .` does not go through the sdist packaging, so `python -m build` and then installing the wheel would seem like the way to go.

Reply:
Probably we want a comment on this bit